pkg/bootstrap/stage.go

```diff
@@ -165,3 +174,27 @@ func extractFromDir(dir, prefix string, img v1.Image, imgName string) error {
 	}
 	return os.Rename(tempDir, dir)
 }
+
+func preloadBootstrapImage(dataDir string) (v1.Image, error) {
+	imagesDir := filepath.Join(dataDir, "agent", "images")
+	files := map[string]os.FileInfo{}
+	if err := filepath.Walk(imagesDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		files[path] = info
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	for fileName := range files {
+		if strings.HasSuffix(fileName, "rke2-runtime-image-amd64.tar") {
```

Comment:
You shouldn't look for a special file.  Just scan all tarballs and call ImageFromPath(..., Tag) with the tag you are looking for.  We expect for airgap the runtime image will be in the same tar as everything else.  You'll get an error if you don't find the tag, just ignore the error.

Reply:
the problem is that image from path requires one image in a tarball